vmdb/spec/controllers/catalog_controller_spec.rb

```diff
@@ -448,4 +448,38 @@
       end
     end
   end
+
+  context "#set_resource_action" do
+    before :each do
+      @st = FactoryGirl.create(:service_template)
+      dialog = FactoryGirl.create(:dialog,
+                                  :label       => "Test Label",
+                                  :description => "Test Description",
+                                  :buttons     => "submit,reset,cancel"
+      )
+      retire_fqname    = 'ns0/cls0/inst0'
+      provision_fqname = 'ns1/cls1/inst1'
+      recon_fqname     = 'ns2/cls2/inst2'
+      edit = {
+        :new          => {
+          :name               => "New Name",
+          :description        => "New Description",
+          :dialog_id          => dialog.id,
+          :reconfigure_fqname => recon_fqname,
+          :retire_fqname      => retire_fqname,
+          :fqname             => provision_fqname},
+      }
+      controller.instance_variable_set(:@edit, edit)
+    end
+    it "saves resource action" do
+      controller.send(:set_resource_action, @st)
+      @st.resource_actions.length.should eq(3)
```

Comment:
This test is not going to catch if we create an invalid resource action if the count happens to match.  I would check that the expected actions are created:
`expect(@st.resource_actions.pluck(:action)).to match_array(%w(Provision Retirement Reconfigure))`